server.py

```diff
@@ -29,6 +29,19 @@ def delete_users(id):
     redis_server.set('users',json_users)
     return reply('', HTTP_204_NO_CONTENT)
 
+@app.route('/users/<id>', methods=['PUT'])
+def update_user(id):
+    global users
+    payload = json.loads(request.data)
+    if users.has_key(id):
+        users[id] = {'name': payload['name'], 'times': payload['times']}
+        message = users[id]
+        rc = HTTP_200_OK
+    else:
+        message = { 'error' : 'User %s was not found' % id }
+        rc = HTTP_404_NOT_FOUND
+    return reply(message, rc)
+
```

Comment:
@gnishma24 this code won't goto redis as you're not updating redis variable. you need to make sure that the changes are pushed to redis. Please make that change